features/include/pcl/features/principal_curvatures.h

```diff
@@ -73,15 +72,18 @@ namespace pcl
       using PointCloudOut = typename Feature<PointInT, PointOutT>::PointCloudOut;
       using PointCloudIn = pcl::PointCloud<PointInT>;
 
-      /** \brief Empty constructor. */
-      PrincipalCurvaturesEstimation () : 
-        xyz_centroid_ (Eigen::Vector3f::Zero ()), 
-        demean_ (Eigen::Vector3f::Zero ()),
-        covariance_matrix_ (Eigen::Matrix3f::Zero ()),
-        eigenvector_ (Eigen::Vector3f::Zero ()),
-        eigenvalues_ (Eigen::Vector3f::Zero ())
+      /** \brief Initialize the scheduler and set the number of threads to use.
+        * \param nr_threads the number of hardware threads to use (0 sets the value to automatic)
+        * \param chunk_size PCL will use dynamic scheduling with this chunk size. Setting it too 
+        *                   low will lead to more parallelization overhead. Setting it too high 
+        *                   will lead to a worse balancing between the threads.
+        */
+      PrincipalCurvaturesEstimation (unsigned int nr_threads = 1, int chunk_size = 256) : 
```

Comment:
```suggestion
      PrincipalCurvaturesEstimation () : 
```

Reply:
This would mean that it is no longer possible to set `nr_threads` in the constructor, and that it is no longer possible to change `chunk_size` at all? Btw the clang-tidy CI already checks use-default-member-init https://github.com/PointCloudLibrary/pcl/blob/master/.clang-tidy#L18

Reply:
Yeah, I forgot that its only the formatting that is not on all files yet.

hmm. Yes, you can change it with the setNumberOfThreads, like the other classes.
But having it as a optional parameter in constructor would be nice as well.